$OpenBSD: patch-src_extensions_dbus_DBusExtension_cpp,v 1.1 2012/10/24 14:57:59 dcoppa Exp $

DBusExtension.cpp: In member function 'void DBusExtension::onAppEvent(int, QVariant)':
DBusExtension.cpp:99: error: expected primary-expression before '{' token
DBusExtension.cpp:99: error: expected `;' before '{' token
DBusExtension.cpp:105: error: expected primary-expression before '{' token
DBusExtension.cpp:105: error: expected `;' before '{' token

--- src/extensions/dbus/DBusExtension.cpp.orig	Wed Oct 24 15:16:26 2012
+++ src/extensions/dbus/DBusExtension.cpp	Wed Oct 24 15:17:00 2012
@@ -96,18 +96,22 @@ void DBusExtension::onAppEvent(int e, QVariant data) {
   switch (e)
     {
     case Event::PlaybackEnded:
-      status = {2,0,0,0};
+      {
+      status = (mpris_status_t){2,0,0,0};
       emit s_statusChanged(status);
+      }
       break;
 
     case Event::PlaybackStarted:
     case Event::TrackChanged:
-      status = {0,0,0,0};
+      {
+      status = (mpris_status_t){0,0,0,0};
       setMetadata(data.value<TrackInfo>());
       emit s_trackChanged(metadata);
       // This is a lie, status doesn't change just because of track change.
       // But last.fm has a bug that prevents PlaybackStarted from being emitted
       emit s_statusChanged(status);
+      }
       break;
 
     default:
